salt/_modules/metalk8s_certs.py

```diff
@@ -0,0 +1,38 @@
+'''
+Various functions for certs generation.
+'''
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'metalk8s_certs'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def compute_subject_alt_name(subject_alt_name):
+    '''
+    Use a dict to compute the suject alternative name string for a certificate
+
+    >>> compute_subject_alt_name({'DNS': ['kubernetes', 'kubernetes.default']})
+    'DNS:kubernetes, DNS:kubernetes.default'
+    >>> compute_subject_alt_name({'DNS': ['kubernetes', 'kubernetes.default'], 'IP Address': ['1.2.3.4', '5.6.7.8']})
```

Comment:
The 'correct' way to declare an IP is not using `IP Address:...`, but `IP:...`. The Salt x509 state takes care of conversion behind the scenes, but it may be easier to do it 'the simple way' from the start.

Reply:
Oh yes, true I will change it